apps/nova/src/components/dashboard/UpcomingLessonsSummaryWrapper.tsx

```diff
@@ -0,0 +1,46 @@
+import { Route } from "@/types/routes";
+import { useUser } from "@litespace/headless/context/user";
+import { useInfiniteLessons } from "@litespace/headless/lessons";
+import { UpcomingLessonsSummary } from "@litespace/luna/Lessons";
+import { ILesson, IUser } from "@litespace/types";
+import dayjs from "dayjs";
+import { useMemo } from "react";
+
+function organizeUpcomingLessons(
+  list: ILesson.FindUserLessonsApiResponse["list"] | null
+) {
+  return (
+    list?.map((item) => ({
+      start: item.lesson.start,
+      tutorName:
+        item.members.find((member) => member.role === IUser.Role.Tutor)?.name ||
+        null,
+      url: Route.Call,
+    })) || []
+  );
+}
+
+export const UpcomingLessonsSummaryWrapper = () => {
+  const { user } = useUser();
+
+  const now = useMemo(() => dayjs.utc().toISOString(), []);
+  const lessonsQuery = useInfiniteLessons({
+    users: user ? [user?.id] : [],
+    userOnly: true,
+    after: now,
+  });
+  const lessons = useMemo(
+    () => organizeUpcomingLessons(lessonsQuery.list),
+    [lessonsQuery.list]
+  );
+  return (
+    <UpcomingLessonsSummary
+      loading={lessonsQuery.query.isLoading || lessonsQuery.query.isPending}
```

Comment:
Let's just show the spinner, incase of the `loading`. We don't want to distrub the layout of the page in case of re-fetching. The re-fetched data can be rendered without a loading indicator. 

```suggestion
      loading={lessonsQuery.query.isLoading}
```


This also applies to other page parts. 
- Chat summary. 
- Student overview.
- Past lessons table.